src/snowflake/snowpark/modin/plugin/_internal/utils.py

```diff
@@ -1995,3 +1996,68 @@ def create_frame_with_data_columns(
 def rindex(lst: list, value: int) -> int:
     """Find the last index in the list of item value."""
     return len(lst) - lst[::-1].index(value) - 1
+
+
+def convert_index_to_qc(index: Any) -> Any:
```

Comment:
I'm not sure how to make the return type "SnowflakeQueryCompiler" without causing circular import issues

Reply:
That does not work either - it still causes the issues

Reply:
```
from typing import TYPE_CHECKING

if TYPE_CHECKING:
    from snowflake.snowpark.modin.plugin.compiler.snowflake_query_compiler import SnowflakeQueryCompiler
```

have you tried this?